tests/test_service.py

```diff
@@ -305,72 +305,118 @@ def test_bad_request(self):
+        ## Best: Promo 5 (FIXED, 150)
+
         product_1 = Product()
-        product_1.id = 123
+        product_1.id = 100
         product_2 = Product()
-        product_2.id = 456
+        product_2.id = 200
+        product_3 = Product()
+        product_3.id = 300
+        product_4 = Product()
+        product_4.id = 400
+
+        db.session.add(product_1)
+        db.session.add(product_2)
+        db.session.add(product_3)
+        db.session.add(product_4)
+
         # Define the promotions
         promotions = [
             {
                 "promo_code": "promo_code_1",
                 "promo_type": PromoType.DISCOUNT,
                 "amount": 40,
                 "is_site_wide": False,
-                "start_date": datetime(2020, 10, 17),
-                "end_date": datetime(2020, 10, 21),
+                "start_date": datetime(2020, 9, 2),
+                "end_date": datetime(2021, 10, 21),
             }, 
             { 
                 "promo_code": "promo_code_2",
                 "promo_type": PromoType.DISCOUNT,
                 "amount": 10,
                 "is_site_wide": True,
-                "start_date": datetime(2020, 10, 21),
-                "end_date": datetime(2020, 10, 23),
+                "start_date": datetime(2020, 8, 21),
+                "end_date": datetime(2021, 10, 23),
             },
             { 
                 "promo_code": "promo_code_3",
                 "promo_type": PromoType.BOGO,
                 "amount": 1,
                 "is_site_wide": False,
-                "start_date": datetime(2020, 10, 16),
-                "end_date": datetime(2020, 12, 30),
+                "start_date": datetime(2020, 9, 1),
+                "end_date": datetime(2021, 5, 30),
             },
             { 
                 "promo_code": "promo_code_4",
                 "promo_type": PromoType.DISCOUNT,
                 "amount": 80,
                 "is_site_wide": False,
                 "start_date": datetime(2020, 10, 14),
-                "end_date": datetime(2020, 10, 18),
+                "end_date": datetime(2021, 5, 18),
+            },
+            { 
+                "promo_code": "promo_code_5",
+                "promo_type": PromoType.FIXED,
+                "amount": 150,
+                "is_site_wide": False,
+                "start_date": datetime(2020, 10, 14),
+                "end_date": datetime(2021, 10, 18),
+            },
+            { 
+                "promo_code": "promo_code_6",
+                "promo_type": PromoType.DISCOUNT,
+                "amount": 80,
+                "is_site_wide": True,
+                "start_date": datetime(2020, 9, 14),
+                "end_date": datetime(2020, 10, 15),
             }
         ]
         tests = [
-            ("123=1000&456=5000", []),
-            ("123=1000&456=5000", [{'123':'promo_code_3'}, {'456': 'promo_code_3'}]),
+            ("100=1000&200=5000", []),
```

Comment:
can you clarify what these query strings mean? is it 1000 of product with id=100 and 5000 of product with id=200 in the cart?

Reply:
@nspektor, yeah, so it is ``` product_id=product_price ```. The **product_price** is of a single product. I'll add this in the comments so that it is understandable in the future. Also, do you think we should include this in the README.md?

Reply:
ohh okay that makes sense, yeah lets include this in the comments and readme